[BUG] fastparquet compatibility tests fail with data mismatch if TZ is not set and system timezone is not UTC

Running the fastparquet compatibility tests will fail if the TZ environment variable is not set.  I suspect this is because after #9652 we can tell the JVM to use a timezone that could be out-of-sync with what the native code thinks the timezone is.

Here's an example of how I reproduced the errors:
```
$ echo $TZ

$ date
Fri 17 Nov 2023 09:36:39 AM CST
$ SPARK_HOME=/home/jlowe/spark-3.4.1-bin-hadoop3/ integration_tests/run_pyspark_from_build.sh -k test_reading_file_written --pyarrow_test
[...]
FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_by_spark_cpu[Timestamp(not_null)0][DATAGEN_SEED=1700171382, INJECT_OOM] - AssertionError: GPU and CPU timestamp values are different at [0, 'a']
FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_gpu[Timestamp(not_null)0][DATAGEN_SEED=1700171382, INJECT_OOM] - AssertionError: GPU and CPU timestamp values are different at [0, 'a']
```

The tests pass if I export TZ=UTC or fail due to a fallback to the CPU (which is appropriate for the current state of the plugin) if I explicitly set TZ=America/Chicago.

--- a/integration_tests/src/main/python/fastparquet_compatibility_test.py
+++ b/integration_tests/src/main/python/fastparquet_compatibility_test.py
@@ -94,6 +94,13 @@ def read_parquet(data_path, local_data_path):
     return read_with_fastparquet_or_plugin
 
 
+def is_timezone_utc():
+    from spark_init_internal import get_spark_i_know_what_i_am_doing
+    import time
+    spark = get_spark_i_know_what_i_am_doing()
+    return spark.conf.get("spark.sql.session.timeZone") == "UTC" and time.tzname[time.daylight] == "UTC"
+
+
 @pytest.mark.skipif(condition=fastparquet_unavailable(),
                     reason="fastparquet is required for testing fastparquet compatibility")
 @pytest.mark.skipif(condition=spark_version() < "3.4.0",
@@ -119,9 +126,12 @@ def read_parquet(data_path, local_data_path):
                  marks=pytest.mark.xfail(reason="fastparquet reads dates as timestamps.")),
     pytest.param(DateGen(nullable=False),
                  marks=pytest.mark.xfail(reason="fastparquet reads far future dates (e.g. year=8705) incorrectly.")),
-    TimestampGen(nullable=False,
-                 start=pandas_min_datetime,
-                 end=pandas_max_datetime),  # Vanilla case.
+    pytest.param(TimestampGen(nullable=False,
+                              start=pandas_min_datetime,
+                              end=pandas_max_datetime),
+                 marks=pytest.mark.skipif(condition=not is_timezone_utc(),
+                                          reason="fastparquet interprets timestamps in UTC timezone, regardless "
+                                                 "of timezone settings")),  # Vanilla case.
     pytest.param(TimestampGen(nullable=False,
                               start=pandas_min_datetime,
                               end=pandas_max_datetime),
@@ -188,9 +198,12 @@ def test_reading_file_written_by_spark_cpu(data_gen, spark_tmp_path):
                  marks=pytest.mark.xfail(reason="fastparquet reads dates as timestamps.")),
     pytest.param(DateGen(nullable=False),
                  marks=pytest.mark.xfail(reason="fastparquet reads far future dates (e.g. year=8705) incorrectly.")),
-    TimestampGen(nullable=False,
-                 start=pandas_min_datetime,
-                 end=pandas_max_datetime),  # Vanilla case.
+    pytest.param(TimestampGen(nullable=False,
+                              start=pandas_min_datetime,
+                              end=pandas_max_datetime),
+                 marks=pytest.mark.skipif(condition=not is_timezone_utc(),
+                                          reason="fastparquet interprets timestamps in UTC timezone, regardless "
+                                                 "of timezone settings")),  # Vanilla case.
     pytest.param(TimestampGen(nullable=False,
                               start=datetime(1, 1, 1, tzinfo=timezone.utc),
                               end=pandas_min_datetime),

--- a/integration_tests/src/main/python/fastparquet_compatibility_test.py
+++ b/integration_tests/src/main/python/fastparquet_compatibility_test.py
@@ -15,6 +15,7 @@
 import pytest
 
 from asserts import assert_gpu_and_cpu_are_equal_collect
+from conftest import is_not_utc
 from data_gen import *
 from fastparquet_utils import get_fastparquet_result_canonicalizer
 from spark_session import spark_version, with_cpu_session, with_gpu_session
@@ -94,13 +95,6 @@ def read_parquet(data_path, local_data_path):
     return read_with_fastparquet_or_plugin
 
 
-def is_timezone_utc():
-    from spark_init_internal import get_spark_i_know_what_i_am_doing
-    import time
-    spark = get_spark_i_know_what_i_am_doing()
-    return spark.conf.get("spark.sql.session.timeZone") == "UTC" and time.tzname[time.daylight] == "UTC"
-
-
 @pytest.mark.skipif(condition=fastparquet_unavailable(),
                     reason="fastparquet is required for testing fastparquet compatibility")
 @pytest.mark.skipif(condition=spark_version() < "3.4.0",
@@ -129,7 +123,7 @@ def is_timezone_utc():
     pytest.param(TimestampGen(nullable=False,
                               start=pandas_min_datetime,
                               end=pandas_max_datetime),
-                 marks=pytest.mark.skipif(condition=not is_timezone_utc(),
+                 marks=pytest.mark.skipif(condition=is_not_utc(),
                                           reason="fastparquet interprets timestamps in UTC timezone, regardless "
                                                  "of timezone settings")),  # Vanilla case.
     pytest.param(TimestampGen(nullable=False,
@@ -201,7 +195,7 @@ def test_reading_file_written_by_spark_cpu(data_gen, spark_tmp_path):
     pytest.param(TimestampGen(nullable=False,
                               start=pandas_min_datetime,
                               end=pandas_max_datetime),
-                 marks=pytest.mark.skipif(condition=not is_timezone_utc(),
+                 marks=pytest.mark.skipif(condition=is_not_utc(),
                                           reason="fastparquet interprets timestamps in UTC timezone, regardless "
                                                  "of timezone settings")),  # Vanilla case.
     pytest.param(TimestampGen(nullable=False,

